logstash-core/lib/logstash/settings.rb

```diff
@@ -0,0 +1,194 @@
+# encoding: utf-8
+
+module LogStash
+  class Settings
+
+    def initialize
+      @settings = {}
+    end
+
+    def register(setting)
+      if @settings.key?(setting.name)
+        raise ArgumentError.new("Setting \"#{setting.name}\" has already been registered as #{setting.inspect}")
```

Comment:
I would define a custom error that inherits from `ArgumentError` It believe it make things easier to debug.